src/tests/test_util.h

```diff
@@ -23,15 +22,15 @@
   }
 #define ASSERT(expr)                                                                       \
   if (!(expr)) {                                                                           \
-    fprintf(stderr, "%s:%d: Assertion '%s' Failed\n", __FILE__, __LINE__, __STRING(expr)); \
+    fprintf(stderr, "%s:%d Assertion '%s' Failed\n", __FILE__, __LINE__, __STRING(expr));  \
     return -1;                                                                             \
   }
 
 #define ASSERT_STRING_EQ(s1, s2) ASSERT(!strcmp(s1, s2));
 
-#define ASSERT_EQUAL_INT(x, y, ...)                                 \
+#define ASSERT_EQUAL(x, y, ...)                                 \
   if (x != y) {                                                     \
-    fprintf(stderr, "%s:%d: ", __FILE__, __LINE__);                 \
+    fprintf(stderr, "%s:%d: Assertion Failed " __VA_ARGS__ ": ", __FILE__, __LINE__);                 \
```

Comment:
I have a newer version of this macro in another branch, that is:

```c
#define ASSERT_EQUAL_INT(x, y, ...)                                 \
  if (x != y) {                                                     \
    fprintf(stderr, "%s:%d: ", __FILE__, __LINE__);                 \
    fprintf(stderr, "%d != %d: " __VA_ARGS__ "\n", (int)x, (int)y); \
    return -1;                                                      \
  }
```

Perhaps to support all scalars, we should print them with `%f` and convert to double. It will print integers just the same. WDYT?

```c
#define ASSERT_EQUAL(x, y, ...)                                 \
  if (x != y) {                                                     \
    fprintf(stderr, "%s:%d: ", __FILE__, __LINE__);                 \
    fprintf(stderr, "%f != %f: " __VA_ARGS__ "\n", (double)x, (double)y); \
    return -1;                                                      \
  }
```